hw/top_earlgrey/syn/chip_earlgrey_asic.sdc

```diff
@@ -283,141 +374,161 @@ set_multicycle_path -hold 1  -end -from [get_clocks SPI_HOST_CLK] -to [get_clock
 
 # computed delays from connected device
 # host in has 2x the pcb delay to account for delays on both outgoing clocks and incoming data
-set SPI_HOST_IN_DEL    [expr 2*${PCB_DEL} + ${STORAGE_OUT_DEL}]
-set SPI_HOST_OUT_DEL   [expr ${PCB_DEL} + ${STORAGE_SETUP_DEL}]
+set SPI_HOST_OUT_DEL_MIN [expr ${STORAGE_HOLD_DEL}  - ${PCB_SKEW}]
+set SPI_HOST_OUT_DEL_MAX [expr ${STORAGE_SETUP_DEL} + ${PCB_SKEW}]
+set SPI_HOST_IN_DEL_MIN  [expr ${STORAGE_OUT_DEL_MIN}]
+set SPI_HOST_IN_DEL_MAX  [expr ${STORAGE_OUT_DEL_MAX} + 2 * ${PCB_DEL}]
+
+# bidir ports, with the downstream device launching on falling edge
+set SPI_HOST_DATA_PORTS [get_ports SPI_HOST_D*]
+set_input_delay -min ${SPI_HOST_IN_DEL_MIN} ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_CLK -add_delay
+set_input_delay -max ${SPI_HOST_IN_DEL_MAX} ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_CLK -add_delay
+set_output_delay -min ${SPI_HOST_OUT_DEL_MIN} \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_CLK -add_delay
+set_output_delay -max ${SPI_HOST_OUT_DEL_MIN} \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_CLK -add_delay
 
-# bidir ports
-set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_CS_L] -clock SPI_HOST_CLK -add_delay
-set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D0]   -clock SPI_HOST_CLK -add_delay
-set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D1]   -clock SPI_HOST_CLK -add_delay
-set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D2]   -clock SPI_HOST_CLK -add_delay
-set_input_delay ${SPI_HOST_IN_DEL} [get_ports SPI_HOST_D3]   -clock SPI_HOST_CLK -add_delay
-
-set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_CS_L] -clock SPI_HOST_CLK -add_delay
-set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D0]   -clock SPI_HOST_CLK -add_delay
-set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D1]   -clock SPI_HOST_CLK -add_delay
-set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D2]   -clock SPI_HOST_CLK -add_delay
-set_output_delay ${SPI_HOST_OUT_DEL} [get_ports SPI_HOST_D3]   -clock SPI_HOST_CLK -add_delay
 
 #####################################
 # SPI DEV clock Passthru Operation  #
 #####################################
 # Passthrough target freq is 33MHz. Using 40MHz to over-constrain
```

Comment:
Oh, I didn't change this. Should we relax this constraint now? 33 MHz is the target we'd _like_ to hit, and setting constraints to 40 MHz seems like major overkill once we're working with real delays.

Reply:
Yeah I think that should be fine, since this will be used with real wire load models now.